src/pivot-table/components/cells/ColumnAdjuster.tsx

```diff
@@ -0,0 +1,77 @@
+import { preventDefaultBehavior } from "@qlik/nebula-table-utils/lib/utils";
+import React, { useEffect, useRef, useState } from "react";
+import { ColumnWidthType } from "../../../types/QIX";
+import type { Cell, DataModel } from "../../../types/types";
+import { GRID_BORDER } from "../../constants";
+import { ColumnWidthValues } from "../../hooks/use-column-width";
+import { CELL_PADDING } from "../shared-styles";
+import { AdjusterBorder, AdjusterHitArea } from "./styles";
+
+const POSITION_ADJUSTMENT = CELL_PADDING + GRID_BORDER;
+
+interface AdjusterProps {
+  cell: Cell;
+  columnWidth: number;
+  dataModel: DataModel | undefined;
+}
+
+/**
+ * Component that is placed on top of column border, to resize the columns.
+ * When you start dragging, mouse move and mouse up listeners are added.
+ * While dragging this components follows the pointer, and on mouse up all column widths are updated.
+ */
+const ColumnAdjuster = ({ cell, columnWidth, dataModel }: AdjusterProps) => {
+  const [internalWidth, setInternalWidth] = useState(columnWidth);
+  const tempWidth = useRef({ initWidth: 0, columnWidth: 0, initX: 0 });
+
+  useEffect(() => setInternalWidth(columnWidth), [columnWidth]);
```

Comment:
This seem like a good use case for `useOnPropsChange` hook (https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes)

Reply:
oh nice